pkg/runtime/wfengine/backendFactory.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package wfengine
+
+import (
+	"time"
+
+	"github.com/dapr/components-contrib/workflows"
+	"github.com/dapr/kit/logger"
+	"github.com/microsoft/durabletask-go/backend"
+	"github.com/microsoft/durabletask-go/backend/sqlite"
+)
+
+const (
+	ActorBackendType  = "workflowbackend.actor"
+	SqliteBackendType = "workflowbackend.sqlite"
+)
+
+type BackendFactory func(appId string, wfe *WorkflowEngine, backendComponentInfo *workflows.WorkflowBackendComponentInfo, log logger.Logger) backend.Backend
+
+var backendFactories = map[string]BackendFactory{
```

Comment:
Is there a way to make this allow some "drop-in" things? Such as what we do in cmd/daprd/components where we can drop a file and it register itself. Otherwise, for each new backend that dtf-go adds, it needs to be added here which is more frail.

Reply:
@ItalyPaleAle Thank you for your suggestion! I appreciate the thought you've put into making the integration more user-friendly. After considering your idea, I wanted to share some thoughts that came to mind.

I really like the simplicity of our current implementation, particularly how adding a new backend only requires a modification in one place, specifically in backend_factory.go. It keeps things neat and focused.

While the idea of a "drop-in" feature is intriguing, I'm concerned that it might introduce additional complexity, especially considering that the number of backends is likely to be much smaller than the variety of dapr components. Keeping the initialization logic within backend_factory.go seems to maintain a straightforward and easy-to-follow structure.

I'm open to further discussion on this, and I'd love to hear your thoughts. Let me know what you think!

Reply:
My concern is the amount of inter-dependency that is being added where there's part of the initialization in this repo, and part in dtf-go.

Ideally all of this should be moved to dtf-go. Dapr will just pass through everything to dtf-go and the initialization will happen in the upstream library. This way, adding a new backend does not require changes to dapr/dapr.